src/data/set/pointwise/smul.lean

```diff
@@ -4,196 +4,389 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Johan Commelin, Floris van Doorn
 -/
 import algebra.module.basic
+import data.list.of_fn
 import data.set.pairwise
 import data.set.pointwise.basic
 
 /-!
-# Pointwise operations of sets
+# Algebraic structure of pointwise operations
 
-This file defines pointwise algebraic operations on sets.
+This file provides lemmas for pointwise operations on `set` in monoids, groups, modules, and shows
+that `set α` is a semigroup/monoid/... if `α` is.
 
 ## Main declarations
 
-For sets `s` and `t` and scalar `a`:
-* `s • t`: Scalar multiplication, set of all `x • y` where `x ∈ s` and `y ∈ t`.
-* `s +ᵥ t`: Scalar addition, set of all `x +ᵥ y` where `x ∈ s` and `y ∈ t`.
-* `s -ᵥ t`: Scalar subtraction, set of all `x -ᵥ y` where `x ∈ s` and `y ∈ t`.
-* `a • s`: Scaling, set of all `a • x` where `x ∈ s`.
-* `a +ᵥ s`: Translation, set of all `a +ᵥ x` where `x ∈ s`.
-
-For `α` a semigroup/monoid, `set α` is a semigroup/monoid.
-
 Appropriate definitions and results are also transported to the additive theory via `to_additive`.
 
 ## Implementation notes
 
-* We put all instances in the locale `pointwise`, so that these instances are not available by
-  default. Note that we do not mark them as reducible (as argued by note [reducible non-instances])
-  since we expect the locale to be open whenever the instances are actually used (and making the
-  instances reducible changes the behavior of `simp`.
+We put all instances in the locale `pointwise`, so that these instances are not available by
+default. Note that we do not mark them as reducible (as argued by note [reducible non-instances])
+since we expect the locale to be open whenever the instances are actually used (and making the
+instances reducible changes the behavior of `simp`.
+
+Given that `set α` is a monoid when `α` is, `n • s`, where `n : ℕ`, is ambiguous between pointwise
+scaling and repeated pointwise addition; the former has `(2 : ℕ) • {1, 2} = {2, 4}`, while the
+latter has `(2 : ℕ) • {1, 2} = {2, 3, 4}`. See note [pointwise nat action].
+-/
```

Comment:
Maybe a `TODO: rename from smul` or similar is appropriate here, in case you forget the follow-up PR.